docs/sources/developers/plugins/extend-the-grafana-ui-with-links-and-commands.md

```diff
@@ -0,0 +1,112 @@
+# Extend the Grafana UI with links and commands
+
+Use the Plugin extensions API with your Grafana App plugins to add links to the Grafana UI. Doing so allows you to direct users to your plugins pages from various "placements" within the Grafana application.
+
+For a plugin to successfully register links it must:
+
+- Be an App plugin.
+- Be [preloaded]({{< relref "./metadata" >}}).
+- Be installed and enabled.
+
+## Available placements within Grafana
+
+A placement is a location within the Grafana application UI where a plugin can insert links or commands. All placements within Grafana start with `grafana/`. The following placements are available:
+
+- `grafana/dashboard/panel/menu`: all panel menu dropdowns found in dashboards
+
+## Add a link extension within Grafana
+
+Here's how you can add a link to the dashboard panel menus in Grafana via your plugin:
+
+Define the link extension in your plugin's `module.ts` file. First, define a new instance of the `AppPlugin` class by using the `configureExtensionLink` method. This method takes an object that describes your link extension, including a `title` property for the link text, a `placement` that tells Grafana where the link should appear, and a `path` for the user to navigate to your plugin.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will navigate the user to the basic app',
+  placement: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+});
+```
+
+That's it! Your link will be displayed in dashboard panel menus. When the user clicks the link, they will be navigated to the path you defined earlier.
+
+_Note: Each plugin is limited to a maximum of two links per placement._
+
+## Add a link extension using context within Grafana
+
+The above example works for simple cases however you may want to act on information from the panel the user is navigating from. This can be achieved by making use of the `configure` property which takes a function and returns an object that consists of a `title` property for the link text, a `path` to navigate the user to your plugin. Alternatively this function may return `undefined` if there's a need to hide the link for certain scenarios.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  placement: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one',
+  configure: (link: AppPluginLinkExtension, context: PanelContext) => {
```

Comment:
I think we should probably update this one, as [this PR](https://github.com/grafana/grafana/pull/64958) is being merged soon.
```suggestion
  configure: (context: PanelContext) => {
```